src/Ltb/Ldap.php

```diff
@@ -481,5 +481,28 @@ function get_first_value($ldap_base, $ldap_scope, $ldap_filter, $attribute): str
 
     }
 
+    /**
+     * test if a DN matches filter, base and scope
+     * @param string $dn: entry DN
+     * @param string $dnAttribute: attribute name containing the DN
+     * @param string $filter
+     * @param string $base
+     * @param string $scope
+     * @return bool: true if DN matches filter, base and scope
+     */
+    public function matchDn($dn, $dnAttribute, $filter, $base, $scope): bool {
+
+        # Build filter
+        $dn_escape = ldap_escape($dn, "", LDAP_ESCAPE_FILTER);
+        $search_filter = '(&' . $filter . '(' . $dnAttribute . '=' . $dn_escape .'))';
```

Comment:
I am not fond of aggregating a new filter, but it's clearly the most optimized method here, so ok.